QLogger.cpp

```diff
@@ -1,266 +1,252 @@
+      }
+      mModuleDest.insert(module, log);
+
+      if (mode != LogMode::Disabled)
+        log->start();
 
       return true;
    }
 
    return false;
 }
 
-bool QLoggerManager::addDestination(const QString &fileDest, const QStringList &modules, LogLevel level)
+bool QLoggerManager::addDestination(const QString &fileDest, const QStringList &modules, LogLevel level,
+                                    const QString &fileFolderDestination, LogMode mode,
+                                    LogFileDisplay fileSuffixIfFull, LogMessageDisplays messageOptions, bool notify)
 {
+   QMutexLocker lock(&mMutex);
    bool allAdded = false;
 
    for (const auto &module : modules)
    {
-      if (!moduleDest.contains(module))
+      if (!mModuleDest.contains(module))
       {
-         const auto log = new QLoggerWriter(fileDest, level);
+          const auto log = new QLoggerWriter((fileDest.isEmpty() ? mDefaultFileDestination : fileDest),
+                                             (level == LogLevel::Warning ? mDefaultLevel : level),
+                                             (fileFolderDestination.isEmpty() ? mDefaultFileDestinationFolder : QDir::fromNativeSeparators(fileFolderDestination)),
+                                             (mode == LogMode::OnlyFile ? mDefaultMode : mode),
+                                             (fileSuffixIfFull == LogFileDisplay::DateTime ? mDefaultFileSuffixIfFull : fileSuffixIfFull),
+                                             (messageOptions.testFlag(LogMessageDisplay::Default) ? mDefaultMessageOptions : messageOptions));
+         log->setMaxFileSize(mDefaultMaxFileSize);
          log->stop(mIsStop);
 
-         moduleDest.insert(module, log);
+         mModuleDest.insert(module, log);
+
+         if (!mIsStop)
+         {
+            if (notify) {
+                const auto threadId
+                        = QString("%1").arg((quintptr)QThread::currentThread(), QT_POINTER_SIZE * 2, 16, QChar('0'));
+                log->enqueue(QDateTime::currentDateTime(), threadId, module, LogLevel::Info, "", "", -1, "Adding destination!");
+            }
+            if (mode != LogMode::Disabled)
+              log->start();
+         }
+
          allAdded = true;
       }
    }
 
    return allAdded;
 }
 
-void QLoggerManager::queueMessage(const QString module, const QVector<QVariant> &logData)
+void QLoggerManager::clearFileDestinationFolder(const QString &fileFolderDestination, int days)
 {
-   if (mNonWriterQueue.count(module) < QUEUE_LIMIT)
-      mNonWriterQueue.insert(module, logData);
+    QDir dir(fileFolderDestination + QStringLiteral("/logs"));
+    if (!dir.exists()) {
+        return;
+    }
+
+    dir.setFilter(QDir::Files | QDir::Hidden | QDir::NoSymLinks);
+    const QFileInfoList list = dir.entryInfoList();
+
+    const QDateTime now = QDateTime::currentDateTime();
+
+    for (int i = 0; i < list.size(); ++i) {
```

Comment:
I believe it's possible to iterate over the list using const iterators.

Reply:
You right, thanks